Update documentation

JakeHartnell opened the pull request:
+ Provide details on how to make a user an admin and access the admin dashboard.
+ Describe how feature flags are used and where they can be enabled
+ Clarify how to view builds of the documentation
+ Reorganize a bit

I was wanting to enable feature flags to review the groups work with @conordelahunty and realized that there was no information about them or making users an administrator in the documentation. This PR provides that info and makes a few other changes which I feel make things clearer.

seanh commented:
Some broken links here:

```
/home/seanh/Projects/hypothesis/h/docs/development_guide/chrome-extension.rst:17: WARNING: unknown document: /hacking/install
/home/seanh/Projects/hypothesis/h/docs/development_guide/chrome-extension.rst:72: WARNING: unknown document: /hacking/ssl
/home/seanh/Projects/hypothesis/h/docs/development_guide/chrome-extension.rst:175: WARNING: unknown document: /hacking/ssl
```

seanh commented:
`README.rst` and `CONTRIBUTING.rst` contain links to hacking that'll be broken too

seanh commented:
I can't remember how it was done, but in CKAN when we moved docs pages around we used to leave redirects behind somehow

seanh commented:
I'd say docs for the adminstration interface belong in a user's guide, not in the development guide.

seanh commented on docs/development_guide/administration.rst:
```diff
@@ -0,0 +1,14 @@
+To access the admin interface, a user must be logged into hypothes.is and have admin permissions.
+To grant admin permissions to a user, run the following command in the virtual environment:
+
+	hypothesis admin <config_uri> <username>
```
This doesn't render properly, you need:

```
  .. code-block:: bash

   hypothesis admin <config_uri> <username>
```

Same for the one below

seanh commented on docs/development_guide/documentation.rst:
```diff
@@ -7,5 +7,7 @@ To build the documentation, ensure that Sphinx_ is installed and issue the
 
-.. _Sphinx: http://sphinx-doc.org/
+You can view the documentation by running a static web server in the newly
+generated ``_build/html/`` directory.
```
Example command?

seanh commented on docs/development_guide/index.rst:
```diff
@@ -1,4 +1,4 @@
-Contributor's guide
+Developer's guide
 ###################
```
Disagree with these. Contributors could (in the future) include designers, documentation writers, translators, QA people and bug reporting, etc etc. The contributor's guide would be the one place to list how-tos for all of these

seanh commented on docs/development_guide/install_development.rst:
```diff
@@ -289,3 +289,11 @@ use and configure it.
+Feature Flags
+-------------
+
+Features under development at hypothesis are hidden behind feature flags. You can enable them
```
Feature flags aren't just for features currently under development. The idea is that every feature (well, most) is behind a feature flag, all flags are turned off by default, and the site admins can choose which features to have on for which users. I don't think we're gonna remove the groups feature flag once groups is done, for example, we'll just turn the flag on for our production deployment.

seanh commented:
> I can't remember how it was done, but in CKAN when we moved docs pages around we used to leave redirects behind somehow

It looks like we just did this: <http://ckan.readthedocs.org/en/latest/api.html>. It's better than nothing, there may be links to hacking out there that we forget to update or that we don't control.

JakeHartnell commented:
I decided the reorganization wasn't really needed, so I changed some things back to how they were. Also, implemented your suggestions.

Thanks for quick and quality comments @seanh! :+1:

seanh commented on docs/hacking/documentation.rst:
```diff
@@ -7,5 +7,13 @@ To build the documentation, ensure that Sphinx_ is installed and issue the
+	
+	$ npm install http-server -g
+	$ cd _build/html
+	$ http-server
```
Come on, these are docs built with Python/Sphinx! :)

```
  .. code-block:: bash
     pushd _build/html/; python -m SimpleHTTPServer; popd
```

JakeHartnell replied:
: P

seanh commented on docs/hacking/administration.rst:
```diff
@@ -0,0 +1,18 @@
+	.. code-block:: bash
+	
+	hypothesis admin <config_uri> <username>
+
```
These are still not rendering correctly, I think you indented them too far. The `.. code-block:: bash` line should be o lines indented, the other one should be 3 lines indented

seanh commented on docs/hacking/install.rst:
```diff
@@ -289,3 +289,10 @@ use and configure it.
+
+Features flags allow admins to enable or disable features for certain groups of
+users. You can enable or disable them from the administration dashboard. Please consult the :doc:`administration` documentation for more information on accessing the admin dashboard.
+
```
Wrap to 79 chars

JakeHartnell commented:
Better?

seanh commented:
Yep :+1:
